rustls/src/server/hs.rs

```diff
@@ -369,7 +384,7 @@ impl State for ExpectClientHello {
             // The SNI hostname is immutable once set.
             assert!(conn.sni.is_none());
             conn.sni = Some(sni.clone());
-        } else if conn.sni != sni {
+        } else if !sni_exact_eq(&conn.sni, &sni) {
```

Comment:
You need to do this comparison before you convert `sni` to `webpki::DnsName`. if it is important that we do this case-sensitively then we should have a test that would have found this issue.

Reply:
I don't know if it is important. What do you think?

Reply:
I don't think this is important, and certainly not important enough to warrant completely fixing this. Ultimately `conn.sni` here is a `webpki:DnsName` and so has already been case-smashed.

Because we only use case-smashed names (eg, for certificate resolution, the public API, etc.), neither the ability to resume as session with a SNI that differs by case, nor varying the case in a HRR would lead to any real problem.

This also wasn't changed in #638.